packages/contracts/contracts/test-helpers/ContextContract.sol

```diff
@@ -121,6 +121,33 @@ contract ContextContract {
         }
     }
 
+    function getCHAINID() public {
+        // bitwise right shift 28 * 8 bits so the 4 method ID bytes are in the right-most bytes
+        bytes32 methodId = keccak256("ovmCHAINID()") >> 224;
+        address addr = executionManagerAddress;
+
+        assembly {
+            let callBytes := mload(0x40)
+            calldatacopy(callBytes, 0, calldatasize)
+
+            // replace the first 4 bytes with the right methodID
+            mstore8(callBytes, shr(24, methodId))
+            mstore8(add(callBytes, 1), shr(16, methodId))
+            mstore8(add(callBytes, 2), shr(8, methodId))
+            mstore8(add(callBytes, 3), methodId)
+
+            // overwrite call params
+            let result := mload(0x40)
+            let success := call(gas, addr, 0, callBytes, calldatasize, result, 500000)
```

Comment:
Looks like there are still a lot of remnants from the old hacky way that I missed in updates, but IIRC the last param to `call` should be 0 instead of that arbitrarily large number.